### ISSUE TITLE
No module named 'PyQt4.QtWebKit' - Vintel won't start on Archlinux

### ISSUE THREAD

**Issue by TheT3ngu** [XanthosX/vintel#159]
Hello everyone,

I'm running an Archlinux-Setup and it seems like with every release of Vintel it's more difficult to get Vintel running. Since ~ 3 months it doesn't start at all. 
Now if I run `python vintel.py`, I get this:

```Traceback (most recent call last):
  File vintel.py, line 31, in <module>
    from vi.ui import viui, systemtray
  File /home/tengu/vintel/src/vi/ui/viui.py, line 34, in <module>
    from PyQt4.QtWebKit import QWebPage
ModuleNotFoundError: No module named PyQt4.QtWebKit
```
Thing is, QtWebKit is installed:
```
pacman -Qs qtwebkit
local/qtwebkit 2.3.4-5
    An open source web browser engine (Qt port)
```
Other dependencies are installed as well as far as I know:
```
local/pyqt4-common 4.12-4
    Common PyQt files shared between python-pyqt4 and python2-pyqt4
local/pyqt5-common 5.8.2-2
    Common PyQt files shared between python-pyqt5 and python2-pyqt5
local/python-pyqt4 4.12-4
    A set of Python 3.x bindings for the Qt toolkit
local/python-pyqt5 5.8.2-2
    A set of Python 3.x bindings for the Qt5 toolkit
local/python-qtpy 1.2.1-1
    Provides an uniform layer to support PyQt5, PyQt4 and PySide with a single codebase
local/python2-pyqt4 4.12-4
    A set of Python 2.x bindings for the Qt toolkit
local/python2-pyqt5 5.8.2-2
    A set of Python 2.x bindings for the Qt5 toolkit
local/qutebrowser 0.10.1-1
    A keyboard-driven, vim-like browser based on PyQt5 and QtWebKit
```
also the following BeautifulSoup:
```
local/python-beautifulsoup4 4.6.0-1
    A Python HTML/XML parser designed for quick turnaround projects like screen-scraping
local/python2-beautifulsoup3 3.2.1-3
    A Python HTML/XML parser designed for quick turnaround projects like screen-scraping
local/python2-beautifulsoup4 4.6.0-1
    A Python HTML/XML parser designed for quick turnaround projects like screen-scraping
```
What can I do at this point? I'm thankful for every hint.

**Comment by RealGrep** [comment]
Yup, QtWebKit was removed from PyQt4 and likely won't return, from what I can tell.

This page has some info on it: https://www.archlinux.org/todo/phasing-out-qtwebkit/

From what I can see, it needs to be moved to PyQt5. Which is probably a good idea, anyway.

I would love to be able to run this, but I can see why the situation is as it is, and I'm disinclined to compile my own version with it enabled. It was removed for a good reason.

However, you could rebuild it yourself using the Arch Build System, but I take no responsibility anything: https://wiki.archlinux.org/index.php/Arch_Build_System

**Comment by TheT3ngu** [comment]
Hey RealGrep,

thanks for the answer. I did some research too and stumbled over news that Arch phased out QtWebKit, but not as clear as in your link.
well, shit then. Just to be sure, I have 2 options right now if I want to use vintel under Arch,
 1) wait until/if vintel is ported to PyQt5
 2) build my own version of PyQt4 with QtWebKit via ABS

- did I get this right?

**Comment by RealGrep** [comment]
Pretty much.

I should point out that there's a port to PyQt5 already, but it's not done and the map flickers rather annoyingly (maybe needs to be double-buffered, who knows). Still, you can find it by clicking on the "Graphs" tab, then "Network". There's a fork by Islay Tzash. Click on the last dot along the timeline and that should bring you to the fork. Check out the proper branch and try that.

There are pull requests from there to this repository, but they've been sitting there since January 18th. I guess the author is busy or something, but doesn't seem like this repository is moving at all, these days, sadly.
